### PR TITLE
Task now fails if OS other than windows

### PR DESCRIPTION
**Pull request by ShreyasRmsft**

### CODE REVIEW

File: Tasks/VsTest/Strings/resources.resjson/en-US/resources.resjson [hunk 1]
```diff
@@ -158,5 +158,6 @@
   "loc.messages.testImpactAndCCWontWork": "Test Impact (Run only Impacted tests) and Code Coverage data collector will not work.",
   "loc.messages.ToolsInstallerInstallationError": "The Visual Studio Test Platform tools installer did not run or did not complete the installation successfully, please refer to the following blog for information on how to use the tools installer: https://aka.ms/vstesttoolsinstaller",
   "loc.messages.OverrideUseVerifiableInstrumentation": "Overriding UseVerifiableInstrumentation field to false in the runsettings file.",
-  "loc.messages.NoTestResultsDirectoryFound": "Test results directory not found."
+  "loc.messages.NoTestResultsDirectoryFound": "Test results directory not found.",
+  "loc.messages.OnlyWindowsOsSupported": "This task is meant to run on Windows machines only. Please retry with a Windows build agent."
```
Comment (nigurr) [inline Tasks/VsTest/Strings/resources.resjson/en-US/resources.resjson]: get this from prachi

File: Tasks/VsTest/runvstest.ts [hunk 2]
```diff
@@ -6,6 +6,13 @@ import * as path from 'path';
 import * as distributedTest from './distributedtest';
 import * as ci from './cieventlogger';
 import * as utils from './helpers';
+import * as os from 'os';
+const osPlat: string = os.platform();
+
+if (osPlat !== 'win32') {
+    // Fail the task if os is not windows
+    tl.setResult(tl.TaskResult.Failed, tl.loc('OnlyWindowsOsSupported'));
```
Comment (nigurr) [inline Tasks/VsTest/runvstest.ts]: did you test it? message shown?
where are you exiting?

File: Tasks/VsTest/Strings/resources.resjson/en-US/resources.resjson (hunk 1 above)
Comment (ShreyasRmsft) [reply Tasks/VsTest/Strings/resources.resjson/en-US/resources.resjson]: have pinged her, waiting for reply

File: Tasks/VsTest/runvstest.ts [hunk 3]
```diff
@@ -6,38 +6,46 @@ import * as path from 'path';
...
-try {
-    tl.setResourcePath(path.join(__dirname, 'task.json'));
-    utils.Helper.setConsoleCodePage();
-    const useDtaExecutionEngine = isDtaEngineRequired();
-    if (useDtaExecutionEngine) {
-        ci.publishEvent({
-            runmode: 'distributedtest', parallelism: tl.getVariable('System.ParallelExecutionType'),
-            testtype: tl.getInput('testSelector')
-        });
+if (osPlat !== 'win32') {
```
Comment (nigurr) [inline Tasks/VsTest/runvstest.ts]: did you evaluate "platforms" option in execution?

Comment (ShreyasRmsft) [reply Tasks/VsTest/runvstest.ts]: i set the platform to linux and it still ran on a windows machine

Comment (ShreyasRmsft) [reply Tasks/VsTest/runvstest.ts]: Will file a bug on them

**Pull request merged**